Zend/zend_compile.c

```diff
@@ -4538,11 +4539,40 @@ void zend_compile_new(znode *result, zend_ast *ast) /* {{{ */
 void zend_compile_clone(znode *result, zend_ast *ast) /* {{{ */
 {
 	zend_ast *obj_ast = ast->child[0];
+	zend_ast *initializer_ast = ast->child[1];
 
 	znode obj_node;
 	zend_compile_expr(&obj_node, obj_ast);
+	zend_op *opline = zend_emit_op(result, ZEND_CLONE, &obj_node, NULL);
 
-	zend_emit_op_tmp(result, ZEND_CLONE, &obj_node, NULL);
+	if (initializer_ast) {
+		zend_ast_list *list = zend_ast_get_list(initializer_ast);
+		uint32_t i;
+
+		for (i = 0; i < list->children; i++) {
+			zend_ast *property_init_expr_ast = list->child[i];
+			ZEND_ASSERT(property_init_expr_ast->kind == ZEND_AST_INITIALIZER_EXPR);
+			zend_string *property_name = zval_make_interned_string(zend_ast_get_zval(property_init_expr_ast->child[0]));
```

Comment:
Since these are all constant strings, can/should this check for duplicates (e.g. check that zend_hash_str_add succeeds, then free the temporary zend_array) and make duplicates an E_COMPILE_ERROR

Reply:
Yeah, that's a nice possibility. For now I wasn't sure if allowing only constants as property names is the desired behavior, or I should make it possible to use expressions as well.